bridge/discord/discord.go

```diff
@@ -376,3 +378,52 @@ func (b *Bdiscord) handleUploadFile(msg *config.Message, channelID string) (stri
 	}
 	return "", nil
 }
+
+// webhookSend send one or more message via webhook, taking care of file
+// uploads (from slack, telegram or mattermost).
+// Returns messageID and error.
+func (b *Bdiscord) webhookSend(msg *config.Message, webhookID, token string) (res *discordgo.Message, err error) {
+	// WebhookParams can have either `Content` or `File`.
+
+	// We can't send empty messages.
+	if msg.Text != "" {
+		res, err = b.c.WebhookExecute(
+			webhookID,
+			token,
+			true,
+			&discordgo.WebhookParams{
+				Content:   msg.Text,
+				Username:  msg.Username,
+				AvatarURL: msg.Avatar,
+			},
+		)
+		if err != nil {
+			b.Log.Errorf("Could not send text (%s) for message %#v: %s", msg.Text, msg, err)
```

Comment:
I think we better also do a `return nil,err` after the log line

Reply:
See my response below. This way of handling errors is well thought on my end. We can covert it to non-named return values, but I’d leave the logic of only returning the text message sending error after handling files as is.

Reply:
Ok, that's fine for me if you're willing to that, refactor with non-named, but keep your logic intact.